File: product-webapp/webapp/src/app/provider/model/ProviderDetails.ts
export enum Role {
  USER = 'USER',
  ADMIN = 'ADMIN',
  PROVIDER = 'PROVIDER',
}

export interface Address {
  street: string;
  city: string;
  zip: number;
  state: string;
}

export interface ProviderDetails {
  userId: string;
  userName: string;
  email: string;
  role: Role;
  address: Address;
  image: any;
}

File: parking-provider-service/src/main/java/com/example/parkingproviderservice/model/ProviderDetails.java
package com.example.parkingproviderservice.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(indexName = "provider_details")
public class ProviderDetails {
	@Id
	private String userId;
	private String userName;
	private String email;
	private Role role;
	private Address address;
	@Field(type = FieldType.Binary, name = "image")
	private byte[] image;
}
